client/src/App/App.tsx

```diff
@@ -4,17 +4,23 @@ import { Router, View, NotFoundBoundary } from 'react-navi';
 
 import { theme } from '../common/theme';
 import routes from './routes';
+import { GlobalProvider } from '../common/contexts/GlobalContext';
+import { SocketProvider } from '../common/contexts/SocketContext';
 
 const App: React.FC = () => (
-    <ThemeProvider theme={theme}>
-        <Router routes={routes}>
-            <Suspense fallback={null}>
-                <NotFoundBoundary render={renderNotFound}>
-                    <View />
-                </NotFoundBoundary>
-            </Suspense>
-        </Router>
-    </ThemeProvider>
+    <SocketProvider>
+        <GlobalProvider>
+            <ThemeProvider theme={theme}>
+                <Router routes={routes}>
+                    <Suspense fallback={null}>
+                        <NotFoundBoundary render={renderNotFound}>
+                            <View />
```

Comment:
I would say that `Socket` and `Global` provider should be lower in the hierarchy tree. Preferably Just above the `View`. Global provider seems a bit complicated and might change a lot causing it to rerender and therefore all of its children. This can cause wasted rendering of ThemeProvider which can be avoided.